ebean-datasource/src/test/java/io/ebean/datasource/test/MultipoolTest.java

```diff
@@ -0,0 +1,123 @@
+package io.ebean.datasource.test;
+
+import io.ebean.datasource.DataSourceBuilder;
+import io.ebean.datasource.DataSourcePool;
+import io.ebean.datasource.DataSourcePoolListener;
+import io.ebean.test.containers.MariaDBContainer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+
+@Disabled("run manually")
+class MultipoolTest {
+
+	private static MariaDBContainer container;
+
+	private static ExecutorService executor;
+
+	@BeforeAll
+	static void before() {
+		container = MariaDBContainer.builder("latest")
+				.dbName("unit")
+				.user("unit")
+				.password("unit")
+				.build();
+
+		container.start();
+
+		executor = Executors.newCachedThreadPool();
+
+	}
+
+	@AfterAll
+	static void after() {
+		executor.shutdownNow();
+	}
+
+	static class PoolManager implements DataSourcePoolListener {
```

Comment:
it is the simplified model of our listener in our application (in the case of DB2 it would allow 500 connections)